gammapy/utils/random/inverse_cdf.py

```diff
@@ -77,3 +83,125 @@ def sample(self, size):
 
         index = index + self.random_state.uniform(low=-0.5, high=0.5, size=index.shape)
         return index
+
+
+class MapEventSampler:
+    """Map event sampler.
+
+    Parameters
+    ----------
+
+    npred_map : `~gammapy.maps.Map`
+            Predicted number of counts map.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+    lc : `~gammapy.time.models.LightCurveTableModel`
+            Input light-curve model of the source, given with columns labelled
+            as "time" (in seconds) and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    phase_lc : `~gammapy.time.models.PhaseCurveTableModel`
+            Input phase-curve model of the source, given with columns labelled
+            as "Phase" and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    tmin : float
+            Start time of the sampling, given in seconds.
+    tmax : float
+            Stop time of the sampling, given in seconds.
+    """
+
+    def __init__(
+        self, npred_map, random_state=0, lc=None, phase_lc=None, tmin=0, tmax=3600
```

Comment:
I think `t_min` and `t_max` should be required arguments. In case no light- or phase curve is given, it will be sampled from a uniform distribution if the temporal model is present, the it is sampled from the model between `t_min` and `t_max`.

Reply:
Done.